config/internal/configsource/component/configsource.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/internal/configsource/configmodels"
+)
+
+// ConfigSource is the interface to be implemented by objects used by the collector
+// to retrieve external configuration information.
+type ConfigSource interface {
+	component.Component
+
+	// BeginSession signals that the ConfigSource is about to be used to inject data into
+	// the configuration. ConfigSource objects can assume that there won't be
+	// concurrent sessions and can use this call according to their needs:
```

Comment:
Can, or should?

Reply:
It would also be worth being explicit about the difference between BeginSession and Start, inherited from component.Component

Reply:
Change the text a bit, let me know if you think it is not clear enough yet.